metis-enrichment/metis-enrichment-common/src/main/java/eu/europeana/enrichment/utils/EntityMergeUtils.java

```diff
@@ -0,0 +1,899 @@
+import eu.europeana.corelib.definitions.jibx.CloseMatch;
+import eu.europeana.corelib.definitions.jibx.Concept;
+import eu.europeana.corelib.definitions.jibx.Date;
+import eu.europeana.corelib.definitions.jibx.DateOfBirth;
+import eu.europeana.corelib.definitions.jibx.DateOfDeath;
+import eu.europeana.corelib.definitions.jibx.DateOfEstablishment;
+import eu.europeana.corelib.definitions.jibx.DateOfTermination;
+import eu.europeana.corelib.definitions.jibx.End;
+import eu.europeana.corelib.definitions.jibx.EuropeanaType;
+import eu.europeana.corelib.definitions.jibx.ExactMatch;
+import eu.europeana.corelib.definitions.jibx.Gender;
+import eu.europeana.corelib.definitions.jibx.HasMet;
+import eu.europeana.corelib.definitions.jibx.HasPart;
+import eu.europeana.corelib.definitions.jibx.Identifier;
+import eu.europeana.corelib.definitions.jibx.InScheme;
+import eu.europeana.corelib.definitions.jibx.IsPartOf;
+import eu.europeana.corelib.definitions.jibx.IsRelatedTo;
+import eu.europeana.corelib.definitions.jibx.Lat;
+import eu.europeana.corelib.definitions.jibx.LiteralType;
+import eu.europeana.corelib.definitions.jibx.NarrowMatch;
+import eu.europeana.corelib.definitions.jibx.Narrower;
+import eu.europeana.corelib.definitions.jibx.Notation;
+import eu.europeana.corelib.definitions.jibx.Note;
+import eu.europeana.corelib.definitions.jibx.PlaceType;
+import eu.europeana.corelib.definitions.jibx.PrefLabel;
+import eu.europeana.corelib.definitions.jibx.ProfessionOrOccupation;
+import eu.europeana.corelib.definitions.jibx.ProxyType;
+import eu.europeana.corelib.definitions.jibx.RDF;
+import eu.europeana.corelib.definitions.jibx.Related;
+import eu.europeana.corelib.definitions.jibx.RelatedMatch;
+import eu.europeana.corelib.definitions.jibx.ResourceOrLiteralType;
+import eu.europeana.corelib.definitions.jibx.SameAs;
+import eu.europeana.corelib.definitions.jibx.TimeSpanType;
+import eu.europeana.corelib.definitions.jibx._Long;
+import eu.europeana.enrichment.api.external.model.Agent;
+import eu.europeana.enrichment.api.external.model.EnrichmentBase;
+import eu.europeana.enrichment.api.external.model.Label;
+import eu.europeana.enrichment.api.external.model.LabelResource;
+import eu.europeana.enrichment.api.external.model.Part;
+import eu.europeana.enrichment.api.external.model.Place;
+import eu.europeana.enrichment.api.external.model.Resource;
+import eu.europeana.enrichment.api.external.model.Timespan;
+import eu.europeana.corelib.definitions.jibx.Concept.Choice;
+import java.util.ArrayList;
+import java.util.List;
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.jibx.runtime.BindingDirectory;
+import org.jibx.runtime.IBindingFactory;
+import org.jibx.runtime.IUnmarshallingContext;
+import org.jibx.runtime.JiBXException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Created by erikkonijnenburg on 28/07/2017.
+ */
+public class EntityMergeUtils {
```

Comment:
There are tons of SonarQube issues with this class. I think it would be good to fix them as this class is quite essential to what we are trying to do here.

Reply:
I fixed them.